openhtf/exe/__init__.py

```diff
@@ -36,57 +39,27 @@ class TestStopError(Exception):
   """Test is being stopped."""
 
 
-class LogSleepSuppress(object): # pylint: disable=too-few-public-methods
-  """Abstraction for supressing stuff we don't care about."""
-
-  def __init__(self):
-    self.failure_reason = ''
-
-  def __enter__(self):  # pylint: disable=invalid-name
-    return self
-
-  def __exit__(self, exc_type, exc_value, exc_tb):  # pylint: disable=invalid-name
-    if exc_type is not None and exc_type is not threads.ThreadTerminationError:
-      # Only log if there is a failure.
-      _LOG.exception(self.failure_reason)
-      time.sleep(1.0)
-    if exc_type is test_state.BlankDutIdError:
-      # Suppress BlankDutIdError, it's likely transient.
-      return True
-    # Raise all other exceptions, we probably care about them.
-    return False
+class UnfinalizedTestError(Exception):
+  """A Test wasn't finalized, likely we missed an abort condition."""
 
 
 # pylint: disable=too-many-instance-attributes
 class TestExecutor(threads.KillableThread):
   """Encompasses the execution of a single test."""
 
-  daemon = True
-
-  FrameworkStatus = Enum('FrameworkStatus',
-                         ['CREATED', 'START_WAIT', 'INITIALIZING', 'EXECUTING',
-                          'STOP_WAIT', 'FINISHING'])
-
-  def __init__(self, test_data, plug_manager, teardown_function=None):
+  def __init__(self, test_data, test_start, teardown_function=None):
     super(TestExecutor, self).__init__(name='TestExecutorThread')
 
-    self._teardown_function = (teardown_function and
-                               openhtf.PhaseInfo.WrapOrCopy(teardown_function))
-
+    self._teardown_function = (
+        teardown_function and openhtf.PhaseInfo.WrapOrCopy(teardown_function))
     self._test_data = test_data
-    self._plug_manager = plug_manager
-    self._test_start = None
+    self._plug_manager = plugs.PlugManager()
+    self._test_start = test_start
     self._lock = threading.Lock()
-    self._status = self.FrameworkStatus.CREATED
 
   def _asdict(self):
```

Comment:
hm, I think `_asdict()` can just go away on this class now, I'll give that a try.